storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/SnowflakeQueryTemplater.java

```diff
@@ -70,30 +70,34 @@ public SnowflakeQueryTemplater(JdbcConnectionProvider connectionProvider) {
     return createEntityTableRowCountQueries;
   }
 
+  // TODO: support filepath as s3 location
   @Override
   protected List<String> createLoadEntityQuery(
-      String destinationTable, String temporaryTable, File filePath) {
+      String destinationTable, String stagingLocation, URI fileUri) {
     List<String> queries = new ArrayList<>();
-    queries.add(
-        String.format("CREATE TABLE %s AS (SELECT * FROM %s);", temporaryTable, destinationTable));
-    queries.add(
+    String csvFormatQuey =
         String.format(
-            "create or replace file format CSV_format type = 'CSV' field_delimiter = ',' skip_header=1;"));
-    queries.add(String.format("create or replace stage my_stage file_format = CSV_format;"));
-    queries.add(String.format("put file://%s @my_stage auto_compress=false;", filePath));
-    String fileName = filePath.getName();
-    queries.add(
+            "create or replace file format CSV_format type = 'CSV' field_delimiter = ',' skip_header=1;");
+    String createStageQuery =
         String.format(
-            "COPY INTO %s FROM '@my_stage/%s' FILE_FORMAT = CSV_format on_error = 'skip_file';",
-            temporaryTable, fileName));
-    queries.add(
-        String.format("INSERT INTO %s SELECT * FROM %s;", destinationTable, temporaryTable));
+            "create or replace stage my_s3_stage\n"
+                + "  storage_integration = s3_int\n"
+                + "  url = '%s';",
+            stagingLocation);
+    // TODO: fileUri must be the in the stagingLocation??
+    String folderPath = fileUri.toString().substring(stagingLocation.length());
+    String copyIntoDestTable =
+        String.format(
+            "COPY INTO %s FROM '@my_s3_stage/%s' FILE_FORMAT = CSV_format on_error = 'skip_file';",
```

Comment:
Instead of creating a stage every time, how about loading directly from the URI https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#loading-files-directly-from-an-external-location